flatsurf/geometry/gl2r_orbit_closure.py

```diff
@@ -657,7 +654,9 @@ def field_of_definition(self):
             GL(2,R)-orbit closure of dimension at least 4 in H_7(4^3, 0) (ambient dimension 17)
             sage: O.field_of_definition() # optional: exactreal, pyflatsurf
             Number Field in c0 with defining polynomial x^2 - 2 with c0 = 1.414213562373095?
+            sage: bound = E.billiard_unfolding_stratum('half-translation', marked_points=True).dimension()
             sage: for decomposition in O.decompositions(1):  # long time, optional: exactreal, pyflatsurf
+            ....:     if O.dimension() == bound: break
```

Comment:
Don't do that (or at least don't do that everywhere). We want to check that all found directions are contained in the target space and we don't get something bigger than expected.

Reply:
Yes, I agree. But we need to break after a few iterations extra iterations at least. Doctests that run for several minutes (on a single thread) are not useful anymore even if marked with long time imho.

Reply:
What about using `islice` from itertools to bound the number of saddle connection used?
```
from itertools import islice
for decomposition in islice(O.decompositions(1), 10):  # long time, optional: pyflatsurf
    O.update_tangent_space_from_flow_decomposition(decomposition)
    print(O.dimension())
```

Reply:
Though, these doctests are intended for being useful for the user. If someone wants to compute orbit closure of her prefered surface, she might prefer to see your version.